.github/workflows/version-check.yml

```diff
@@ -15,16 +15,27 @@ jobs:
         run: |
           GRADLE_VERSION=$(awk -F= 'NR==1 {print $2}' ./service/gradle.properties)
           NPM_VERSION=$(jq -r '.version' ./client/package.json)
+
+          echo "Gradle version: '$GRADLE_VERSION'."
+          echo "NPM version:    '$NPM_VERSION'."
+
           if [ "$GRADLE_VERSION" != "$NPM_VERSION" ]; then
             echo "Version mismatch between client and service. Update the version in both modules."
             exit 1
           fi
 
       - name: Check version equality between branches
         run: |
+          WORK_VERSION=$(awk -F= 'NR==1 {print $2}' ./service/gradle.properties)
+
           git fetch origin main
+          git checkout main
           MAIN_VERSION=$(awk -F= 'NR==1 {print $2}' ./service/gradle.properties)
-          if [ "$MAIN_VERSION" == "$GRADLE_VERSION" ]; then
+
+          echo "Gradle version on working branch: '$WORK_VERSION'."
+          echo "Gradle version on main branch:    '$MAIN_VERSION'."
+
+          if [ "$WORK_VERSION" == "$MAIN_VERSION" ]; then
             echo "Version number has not changed. Update the version before merging."
             exit 1
           fi
```

Comment:
This is what you get for using AI to generate code and not checking if it even works.